lib/convection/control/cloud.rb

```diff
@@ -114,6 +114,18 @@ def diff(to_stack, options = {}, &block)
           return
         end
 
+        errors = []
+        stacks.each_value do |stack|
+          if stack.error?
+            errors << stack.errors.collect { |x| x.exception.message }
```

Comment:
This is because the different method of error handling. With converges we pass back the error object for the error encountered during the converge. For diffs we are looking for any errors that have already happened and bail if there were any. I guess to say, if the diff caused the error I would take the same approach as converge but as it has already happened we can just collect all the errors and return them. I plan on taking the same approach as i did with diffs to fix converges when checking for initialization errors.